projects/graph/src/draw.py

```diff
@@ -1,14 +1,53 @@
-"""
-General drawing methods for graphs using Bokeh.
-"""
+import math
 
 from bokeh.io import show, output_file
 from bokeh.plotting import figure
-from bokeh.models import (GraphRenderer, StaticLayoutProvider, Circle, LabelSet,
-                          ColumnDataSource)
+from bokeh.models import GraphRenderer, StaticLayoutProvider, Circle
+from bokeh.palettes import Spectral8
+from graph import Graph
 
+graph = Graph()  # Instantiate your graph
+graph.add_vertex('0')
+graph.add_vertex('1')
+graph.add_vertex('2')
+graph.add_vertex('3')
+graph.add_edge('0', '1')
+graph.add_edge('0', '3')
+print(graph.vertices)
 
-class BokehGraph:
-    """Class that takes a graph and exposes drawing methods."""
-    def __init__(self):
-        pass  # TODO
```

Comment:
Your code draws a graph but the BokehGraph class should be implemented with Graph (from graph.py) as an object. I think it is just plotting the vertices and edges above and the code below is not being implemented at all. Is this in progress?